src/mongocxx/collection.cpp

```diff
@@ -414,6 +414,16 @@ std::int64_t collection::count(bsoncxx::document::view filter, const options::co
         rp_ptr = options.read_preference()->_impl->read_preference_t;
     }
 
+    if (options.hint()) {
+        auto filter_with_hint = bsoncxx::builder::stream::document{}
+                                << options.hint().value()
```

Comment:
I've been doing `*options.hint()` instead of `options.hint().value()`. Not sure if there's a difference, but if there isn't, I think the `*options.hint()` option is more common.